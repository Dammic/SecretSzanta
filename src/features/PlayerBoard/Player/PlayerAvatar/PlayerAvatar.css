@import 'variables/variables.css';

@keyframes breathing {
  0% {
    transform: scale(0.8);
  }

  25% {
    transform: scale(0.9);
  }

  60% {
    transform: scale(1);
  }

  80% {
    transform: scale(0.9);
  }

  100% {
    transform: scale(0.8);
  }
}

.avatarWrapper {
    width: var(--player-image-width);
    max-width: 130px;
    position: relative;
    &:hover .fascistPortrait {
        opacity: 0.1; 
    }
}

.portrait {
    width: inherit;
    max-width: inherit;
}

.fascistPortrait {
    position: absolute;
    height: 100%;
    width: 100%;
    transition: all 0.6s;
    z-index: 1;
}

.selectingWaitIcon {
    color: var(--selecting-wait-icon-color);
    position: absolute;
    right: 5px;
    bottom: 5px;
    font-size: 1.5em;
    z-index: 20;
    animation: breathing 3s ease-out infinite normal;
}
.ownerIcon {
    position: absolute;
    top: -3%;
    left: 9%;
    transform: scale(2.5, 2.5);
    z-index: 20;

    color: var(--player-avatar-owner-icon);
}

.dead {
    &:before, &:after {
        position: absolute;
        content: '';
        background: var(--player-cross);
        display: block;
        width: 133%;
        height: 5px;
        transform: rotate(-49deg);
        left: -17%;
        right: 0;
        top: 0;
        bottom: 0;
        margin: auto;
        z-index: 2;
    }
    &:after {
        transform: rotate(49deg);
    }
    & .portrait {
        filter: grayscale(100%);
    }
}

