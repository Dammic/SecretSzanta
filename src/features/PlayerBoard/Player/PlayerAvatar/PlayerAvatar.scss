.avatar-wrapper {
    position: relative;
    .selecting-wait-icon {
        color: $selecting-wait-icon-color;
        position: absolute;
        right: 5px;
        bottom: 5px;
        font-size: 1.5em;
        z-index: 20;
        animation: breathing 3s ease-out infinite normal;
    }
    &.dead {
        &:before, &:after {
            position: absolute;
            content: '';
            background: $player-cross;
            display: block;
            width: 133%;
            height: 5px;
            transform: rotate(-49deg);
            left: -17%;
            right: 0;
            top: 0;
            bottom: 0;
            margin: auto;
            z-index: 2;
        }
        &:after {
            transform: rotate(49deg);
        }

        .portrait {
            filter: grayscale(100%);
        }

    }

    &:hover {
        .portrait {
            &.facist-portrait {
                opacity: 0.1; 
            }
        }
    }
    .portrait {
        width: $player-image-width;
        max-width: 130px;
        &.facist-portrait {
            position: absolute;
            height: 100%;
            width: 100%;
            transition: all 0.6s;
            z-index: 1;
        }
    }
}
