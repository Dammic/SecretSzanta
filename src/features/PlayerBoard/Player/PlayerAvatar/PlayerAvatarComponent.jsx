import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const PlayerAvatarComponent = ({
    liberalAvatar,
    facistAvatar,
    isDead,
    isPlayerWaitedFor,
    className,
}) => {
    const liberalAvatarPicture = require(`../../../../static/Avatar${liberalAvatar}.png`)
    const facistAvatarPicture = facistAvatar ? require(`../../../../static/Avatar${facistAvatar}.png`) : ''

    return (
        <div className={classNames('avatar-wrapper', className, { dead: isDead })}>
            {isPlayerWaitedFor && <i className="fa fa-clock-o selecting-wait-icon" aria-hidden="true" />}
            {facistAvatar && <img className="portrait facist-portrait" src={facistAvatarPicture} alt="Player facist avatar" />}
            <img className="portrait" src={liberalAvatarPicture} alt="Player liberal avatar" />
        </div>
    )
}

PlayerAvatarComponent.propTypes = {

}
export default PlayerAvatarComponent
