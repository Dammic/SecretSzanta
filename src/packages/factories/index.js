export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
