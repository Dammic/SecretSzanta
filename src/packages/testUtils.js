import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
