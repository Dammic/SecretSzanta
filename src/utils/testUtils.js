import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
