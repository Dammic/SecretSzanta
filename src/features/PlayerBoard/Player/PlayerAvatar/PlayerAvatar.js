import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { getAvatar } from '../../../../utils/avatarsHelper'
import { Icon } from '../../../Shared/Icon'
import { PlayerRole } from './PlayerRole/PlayerRole'

import styles from './PlayerAvatar.css'

const PlayerAvatar = ({
    liberalAvatar,
    fascistAvatar,
    isDead,
    isPlayerWaitedFor,
    isOwner,
    role,
    className,
}) => {
    const liberalAvatarPicture = getAvatar(`liberal-${liberalAvatar}`)
    const fascistAvatarPicture = getAvatar(`fascist-${fascistAvatar}`)

    if (!liberalAvatarPicture) {
        return null
    }

    return (
        <div className={classNames(styles.avatarWrapper, className, { [styles.dead]: isDead })}>
            {isOwner && <Icon name="fa-bolt" className={styles.ownerIcon} />}
            {isPlayerWaitedFor && <Icon name="fa-clock-o" className={styles.selectingWaitIcon} />}
            {fascistAvatar && <img className={classNames(styles.portrait, styles.fascistPortrait)} src={fascistAvatarPicture} alt="Player fascist avatar" />}
            <img className={styles.portrait} src={liberalAvatarPicture} alt="Player liberal avatar" />
            {!isDead && <PlayerRole role={role} />}
        </div>
    )
}

PlayerAvatar.displayName = 'PlayerAvatar'
PlayerAvatar.propTypes = {
    liberalAvatar: PropTypes.number.isRequired,
    fascistAvatar: PropTypes.number,
    isDead: PropTypes.bool,
    role: PropTypes.string,
    isPlayerWaitedFor: PropTypes.bool,
    isOwner: PropTypes.bool,
    className: PropTypes.string,
}
export default PlayerAvatar
