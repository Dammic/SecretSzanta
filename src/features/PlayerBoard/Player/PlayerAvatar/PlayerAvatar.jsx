import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { getAvatar } from '../../../../utils/avatarsHelper'
import { Icon } from '../../../Shared/Icon'

import styles from './PlayerAvatar.css'

const PlayerAvatar = ({
    liberalAvatar,
    fascistAvatar,
    isDead,
    isWaitIconVisible,
    isOwner,
    className,
}) => {
    const liberalAvatarPicture = getAvatar(`liberal-${liberalAvatar}`)
    const fascistAvatarPicture = getAvatar(`fascist-${fascistAvatar}`)

    if (!liberalAvatarPicture) {
        return null
    }

    return (
        <div className={classNames(styles.avatarWrapper, className, { [styles.dead]: isDead })}>
            {isOwner && <Icon name="fa-bolt" className={styles.ownerIcon} />}
            {isWaitIconVisible && <Icon name="fa-clock-o" className={styles.selectingWaitIcon} />}
            {fascistAvatar && <img className={classNames(styles.portrait, styles.fascistPortrait)} src={fascistAvatarPicture} alt="Player fascist avatar" />}
            <img className={styles.portrait} src={liberalAvatarPicture} alt="Player liberal avatar" />
        </div>
    )
}

PlayerAvatar.displayName = 'PlayerAvatar'
PlayerAvatar.propTypes = {
    liberalAvatar: PropTypes.number.isRequired,
    fascistAvatar: PropTypes.number,
    isDead: PropTypes.bool,
    isWaitIconVisible: PropTypes.bool,
    isOwner: PropTypes.bool,
    className: PropTypes.string,
}
export default PlayerAvatar
