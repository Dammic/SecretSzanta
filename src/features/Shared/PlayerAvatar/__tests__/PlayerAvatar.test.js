import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { PlayerRole as PlayerRoleDict } from '../../../../../Dictionary'
import { PlayerAvatar } from '../PlayerAvatar'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        liberalAvatar: 1,
        fascistAvatar: null,
        isDead: false,
        isPlayerWaitedFor: false,
        isOwner: false,
        className: 'mockClassName',
        role: PlayerRoleDict.ROLE_CHANCELLOR,
        ...propsOverrides,
    }
    const component = renderMethod(<PlayerAvatar {...props} />)
    return { props, component }
}

describe('<PlayerAvatar />', () => {
    it('returns null if liberalAvatarPicture doesnt exist', () => {
        const { props } = setupProps({ liberalAvatar: 999 })
        expectShallowMatchingSnapshot(<PlayerAvatar {...props} />)
    })

    it('shows liberal avatar, fascist avatar, owner icon, waited for icon, dead classname and role ribbon for full data', () => {
        const { props } = setupProps({ fascistAvatar: 21, isDead: true, isPlayerWaitedFor: true, isOwner: true })
        expectShallowMatchingSnapshot(<PlayerAvatar {...props} />)
    })

    it('doesnt show fascist avatar, owner icon, waited for icon, dead classname for minimum data', () => {
        const { props } = setupProps({ fascistAvatar: null, isDead: false, isPlayerWaitedFor: false, isOwner: false })
        expectShallowMatchingSnapshot(<PlayerAvatar {...props} />)
    })

    it('doesnt show role ribbon if user is dead', () => {
        const { props } = setupProps({ isDead: true })
        expectShallowMatchingSnapshot(<PlayerAvatar {...props} />)
    })
})
