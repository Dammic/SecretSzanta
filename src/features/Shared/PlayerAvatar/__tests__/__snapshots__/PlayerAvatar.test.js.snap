// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerAvatar /> doesnt show fascist avatar, owner icon, waited for icon, dead classname for minimum data 1`] = `
<div
  className="avatarWrapper mockClassName"
>
  <img
    alt="Player liberal avatar"
    className="portrait"
    src="test-file-stub"
  />
  <PlayerRole
    role="ROLE_CHANCELLOR"
  />
</div>
`;

exports[`<PlayerAvatar /> doesnt show role ribbon if user is dead 1`] = `
<div
  className="avatarWrapper mockClassName dead"
>
  <img
    alt="Player liberal avatar"
    className="portrait"
    src="test-file-stub"
  />
</div>
`;

exports[`<PlayerAvatar /> returns null if liberalAvatarPicture doesnt exist 1`] = `null`;

exports[`<PlayerAvatar /> shows liberal avatar, fascist avatar, owner icon, waited for icon, dead classname and role ribbon for full data 1`] = `
<div
  className="avatarWrapper mockClassName dead"
>
  <Icon
    className="ownerIcon"
    name="fa-bolt"
  />
  <Icon
    className="selectingWaitIcon"
    name="fa-clock-o"
  />
  <img
    alt="Player fascist avatar"
    className="portrait fascistPortrait"
    src="test-file-stub"
  />
  <img
    alt="Player liberal avatar"
    className="portrait"
    src="test-file-stub"
  />
</div>
`;
